ClientApp/src/components/Account/signupForm.js

```diff
@@ -0,0 +1,92 @@
+import React, { useContext, useState } from "react";
+import {
+  BoldLink,
+  BoxContainer,
+  FormContainer,
+  Input,
+  SubmitButton,
+  SmallText,
+} from "./common";
+import { Marginer } from "./marginer";
+import { LoginContext } from "./LoginContext";
+import { useForm } from "react-hook-form";
+
+export function SignupForm(props) {
+  
+  const { switchToSignin } = useContext(LoginContext);
+  const { register, handleSubmit } = useForm([]);
+  const { dublicateFound, setDublicateFound } = useState(false);
+
+  const onSignUp = data => {
+    const response = fetch('user/signup', {
+        method: 'POST',
+        headers: { 'Content-type': 'application/json' },
+        body: JSON.stringify(data),
+    });
```

Comment:
You should await fetch here and make this method async or do it with fetch(...).then(response=> { rest of the logic }),